rust/utils/run-locally/src/invariants.rs

```diff
@@ -93,6 +97,19 @@ pub fn termination_invariants_met(
         log_counts.get(LOOKING_FOR_EVENTS_LOG_MESSAGE).unwrap() > &0,
         "Didn't find any logs about looking for events in index range"
     );
+    let total_tx_id_log_count = log_counts.get(TX_ID_INDEXING_LOG_MESSAGE).unwrap();
+    assert!(
+        // there are 3 txid-indexed events:
+        // - relayer: merkle insertion and gas payment
+        // - scraper: gas payment
```

Comment:
`total_tx_id_log_count` just looks at the relayer right?

if we expect a merkle insertion and at least one gas payment, should we make this `*total_tx_id_log_count as u64 >= config.kathy_messages * 2` ?

Reply:
you're right that it does only use relayer logs, but some logs cover multiple events when batches are sent - e.g. `Found log(s) for tx id num_logs = 2`. So there's a chance there are fewer `total_tx_id_log_count` than total messages, if `batch_count * 2 < message_count`

Reply:
ah I see

I guess this could this be flaky then / it's just a soft check that things work?

Reply:
could be flaky and is more of a soft check, if it fail randomly I'll just change it to be nonzero